### PR TITLE
SWI-6546

### PR DESCRIPTION
**Pull request by ajrice6713**
Don't throw exceptions when there is a success status code but errors in the body

### CODE REVIEW

**Comment by ajrice6713** [comment]
Fixes #83

**Review by brianluisgomez — APPROVED** [review]

File: Bandwidth.Iris/Client.cs [hunk 1]
```diff
@@ -396,51 +397,50 @@ private static string FixPath(string path)
...
                             {
-                                throw new AggregateException(exceptions);
+                                code = error.Element("Code");
+                                description = error.Element("Description");
                             }
-                            code = doc.Descendants("resultCode").FirstOrDefault();
-                            description = doc.Descendants("resultMessage").FirstOrDefault();
                         }
-                        else
+                        if (code != null && description != null && !string.IsNullOrEmpty(code.Value) && code.Value != "0")
```
Comment (tmonck) [inline Bandwidth.Iris/Client.cs:433]: is `code != null` necessary in this block since the previous check is `code == null`?

**Review by tmonck — APPROVED** [review]

File: Bandwidth.Iris/Client.cs (hunk 1 above)
Comment (ajrice6713) [reply Bandwidth.Iris/Client.cs:433]: Probably not 😂

**Review by tmonck — APPROVED** [review]

File: Bandwidth.Iris/Client.cs [hunk 2]
```diff
@@ -396,51 +397,50 @@ private static string FixPath(string path)
 
         private async Task CheckResponse(HttpResponseMessage response)
         {
-            try
+            if (!response.IsSuccessStatusCode)
```
Comment (hamermike) [inline Bandwidth.Iris/Client.cs:400]: You can avoid the additional nesting with a return at the top.

`if (response.IsSuccessStatusCode)
 return;`

**Review by brianluisgomez — APPROVED** [review]

**Review by hamermike — APPROVED** [review]

**Pull request merged**